OpenImageIOReader : Don't leak tileBatchIndex context to input plugs

danieldresser-ie opened the pull request:
Fairly straightforward fix with corresponding test.

Note that I've removed the use of a GlobalScope - as far as I can tell, whenever tileBatch is evaluated, it's already within a global scope, though it would be good for you to double check this.

johnhaddon commented on python/GafferImageTest/OpenImageIOReaderTest.py:
```diff
@@ -515,6 +515,19 @@ def testOffsetAlignment( self ) :
+		s["expression"] = Gaffer.Expression()
+		s["expression"].setExpression( 'parent["reader"]["fileName"] = "%s"' % self.fileName )
+
+		with Gaffer.ContextMonitor( root = s["expression"]["out"] ) as cm :
```
Looks like we have a trivial CI error :

```
Error: ERROR: testFileNameContext (GafferImageTest.OpenImageIOReaderTest.OpenImageIOReaderTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/__w/gaffer/gaffer/build/python/GafferImageTest/OpenImageIOReaderTest.py", line 526, in testFileNameContext
    with Gaffer.ContextMonitor( root = s["expression"]["out"] ) as cm :
KeyError: "'out' is not a child of 'expression'"
```

Not sure how that could have passed at your end - perhaps you have a backwards compatibility shim for `out`? I think the simplest fix is just `root = s["expression"]`.

johnhaddon commented on Changes.md:
```diff
@@ -1,3 +1,11 @@
+Fixes
+-----
+￼
+- ImageReader : Fix fileName being evaluated in multiple contexts
```
- ImageReader : Fixed bug which caused `fileName` to be evaluated in multiple contexts.

danieldresser-ie commented:
Addressed feedback.  Github reported a conflict in the Changes, so I merged it using the github UI ... is that acceptable, there doesn't seem to be any way to squash it without going back and reuploading?

johnhaddon commented:
We tend to rebase and push to fix the conflicts, as we prefer the cleaner history you get. I know that's a pain from behind a firewall though, so it's fine to leave that to me as a final merge step. I'll do that now.
